addons/stock/models/stock_orderpoint.py

```diff
@@ -320,27 +320,39 @@ def _search_qty_to_order(self, operator, value):
-                qty_forecast_with_visibility = orderpoint.product_id.with_context(product_context).read(['virtual_available'])[0]['virtual_available'] + orderpoint._quantity_in_progress()[orderpoint.id]
-                qty_to_order = max(orderpoint.product_min_qty, orderpoint.product_max_qty) - qty_forecast_with_visibility
-                remainder = orderpoint.qty_multiple > 0.0 and qty_to_order % orderpoint.qty_multiple or 0.0
-                if (float_compare(remainder, 0.0, precision_rounding=rounding) > 0
-                        and float_compare(orderpoint.qty_multiple - remainder, 0.0, precision_rounding=rounding) > 0):
-                    if float_is_zero(orderpoint.product_max_qty, precision_rounding=rounding):
-                        qty_to_order += orderpoint.qty_multiple - remainder
-                    else:
-                        qty_to_order -= remainder
+            return (
+                orderpoint.product_id
+                and orderpoint.location_id
+                and float_compare(orderpoint.qty_forecast, orderpoint.product_min_qty, precision_rounding=rounding) < 0
+            )
+
+        orderpoints = self.filtered(to_compute)
+        orderpoints_by_location = {}
+        for location, grouped_orderpoints in orderpoints.grouped(lambda o: o.location_id).items():
+            orderpoints_by_location[location] = grouped_orderpoints
+        qty_in_progress_by_orderpoint = {}
+        for orderpoint_group in orderpoints_by_location.values():
+            qty_in_progress_by_orderpoint.update(orderpoint_group._quantity_in_progress())
+        for orderpoint in orderpoints:
+            qty_to_order = 0.0
+            rounding = orderpoint.product_uom.rounding
+            # We want to know how much we should order to also satisfy the needs that gonna appear in the next (visibility) days
+            product_context = orderpoint._get_product_context(visibility_days=orderpoint.visibility_days)
+            qty_forecast_with_visibility = orderpoint.product_id.with_context(product_context).read(['virtual_available'])[0]['virtual_available'] + qty_in_progress_by_orderpoint[orderpoint.id]
+            qty_to_order = max(orderpoint.product_min_qty, orderpoint.product_max_qty) - qty_forecast_with_visibility
+            remainder = orderpoint.qty_multiple > 0.0 and qty_to_order % orderpoint.qty_multiple or 0.0
+            if (float_compare(remainder, 0.0, precision_rounding=rounding) > 0
+                    and float_compare(orderpoint.qty_multiple - remainder, 0.0, precision_rounding=rounding) > 0):
+                if float_is_zero(orderpoint.product_max_qty, precision_rounding=rounding):
+                    qty_to_order += orderpoint.qty_multiple - remainder
+                else:
+                    qty_to_order -= remainder
             orderpoint.qty_to_order_computed = qty_to_order
+        (self - orderpoints).qty_to_order_computed = False
```

Comment:
```suggestion
        (self - orderpoints).qty_to_order_computed = 0
```
Since the field is a `Float`